libavoidrepetition.h

```diff
@@ -39,10 +39,31 @@
 //add more data types at will when needed.
 //we're gonna use `TYPE_dataTypeName` anyway.
 
-#define type(var) _Generic(var, \
-	int: TYPE_INT, char: TYPE_CHAR, char*: TYPE_STR, \
-	long: TYPE_LONG, double: TYPE_DOUBLE, float: TYPE_FLOAT,\
-	default: TYPE_UNKNOWN)
+void ret_char(){printf("Char\n");}
```

Comment:
Well, please tell me how can you use `printf` in compile-time code? If you use`printf`, your code itself can't reuse `type()` within itself. Please think through.

Reply:
It may be because I'm not a native english speaker but I didn't really understand you said.

Reply:
``` your code itself can't reuse type() within itself. Please think through.``` didn't really understand that part.

Reply:
Oh. More simply, if the `type()` implementation uses `printf()`, compile-time code won't be able to use `type()`. Like, `if (type(i)==TYPE_INT) {...}`

Reply:
we could just make the functions return a string, and strcmp that, or like u did before, make them return a number. Want me to do that ?

Reply:
If you return a string, we're gonna be needing to `strcmp()` that. That's troublesome. So, please, just keep my implementation of `type()`. Is there any problem with my implementation of `type()`? If yes, feel free to tell me aloud.

Reply:
You wrote in the readme that type was yet to be implemented and I thought that your code didn't work, didn't bother to check. :|

Reply:
is the sorting function alright tho ?

Reply:
Oh, I'm so sorry. Forgot to update thr README. Thanks!

Reply:
Well no proble, but could you give me some advice on the sorting algorithm, I know its not the fastests, but it's faster than bubble sort and easier to implement than merge sort so idk

Reply:
Okay, updated the README. (Merge the commit into your fork!)

Python uses a stable & robust algorithm `timsort`. Can you work on implementing `timsort`? I'm very invested on implementing `print()`.

Reply:
okay ima start working on timsort, but i'm in class rn so in 30 min i'll start

Reply:
Done ^_^.

Reply:
Would you like to implement the following under `index()` too? Supporting multiple data types, just like Python3?

- Given any array, `index()` returns position of any element. (`//TODO: @koreaneggroll`)
- Given any string, return position of another string. (Actually, I already implemented it. I'll just commit it later) (`//DONE: @midnqp`)

Reply:
Alright, on it

Reply:
wait, in python the .index() function works only on strings aparently. Tested it in a file. Gives error: ```Traceback (most recent call last):
  File "test.py", line 2, in <module>
    print(test.index('1'))
AttributeError: 'int' object has no attribute 'index'```

Reply:
Yeah. Only arrays & strings.

Reply:
Couldn't get the array index working tbh, idk what i did wrong but doesn't work

Reply:
Perfectly normal! I'll take it forward from what you've proceeded. Just commit it!

Reply:
I just deleted the code for the array since it wasn't either fast nor working correctly, the code for the index of a string is already commited here tho